app/lib/tenejo/pf_object.rb

```diff
@@ -14,8 +14,9 @@ def initialize(h, lineno)
       self.lineno = lineno
       h.delete(:object_type)
       h.each do |k, v|
-        if respond_to?("#{k}=")
-          send("#{k}=", v) if v.present?
+        if respond_to?("#{k}=") && v.present?
+          v = v.split("|~|") if v =~ /\|~\|/
```

Comment:
what about 
v = v.split("|~|")
if should_be_singular?(k)
    rec.warnings[k] << "#{k} on line #{rec.lineno} has extra values that will be ignored" if v.count > 1
    v = v.first
end